### PR TITLE
Added 'complete' installation with extras_require

### PR DESCRIPTION
**Pull request by alejoe91**
@samuelgarcia not sure if we should fix the versions or have `>=`. What do you think?

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi Alessio.
I would prefer a "full" over "complete".
This is because I would like to implement for the next release:
spikeinterface/full.py that import everything (extractors/toolkit/...).

So we could do
```python
# only the core
import spikeinterface as si
si.only_core_func()

# all subpackage
import spikeinterface.full as si
si.every_func_in_any_sub_module()
```

I will do the same for big_rafcatoring.

**Comment by samuelgarcia** [comment]
And I prefer >=

File: complete_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+h5py>=3.2.1
```
Comment (samuelgarcia) [inline complete_requirements.txt:1]: Forcing h5py to version 2 is dangerous today. We will break lot of stuff.

**Comment by samuelgarcia** [comment]
@alejoe91 : tell when ready to merge to old branch.

**Comment by samuelgarcia** [comment]
yep. tell me when we can merge

**Comment by samuelgarcia** [comment]
Is this ready to merge ?

**Comment by samuelgarcia** [comment]
Ready to merge ?

**Comment by alejoe91** [comment]
Yes!

**Pull request merged**